pkg/build/api/types.go

```diff
@@ -228,8 +228,58 @@ type BuildStatus struct {
 
 	// Output describes the Docker image the build has produced.
 	Output BuildStatusOutput
+
+	// Steps contains details about each build step
+	Steps Steps
+}
+
+// Steps is a slice of build steps that have been recorded.
+type Steps []StepInfo
+
+// RecordInfo records details about each build step
+func (s *Steps) RecordInfo(stepName StepName, startTime unversioned.Time, stopTime unversioned.Time) {
```

Comment:
> @bparees nothing comes to mind except convention (see types.go in kubernetes). @deads2k?

To my knowledge, convention.  The structs in types.go are supposed to be pure value objects without any higher order logic on them.  Doing it that way makes it possible to separate the type definitions packages from any logic performed on them.

@smarterclayton smacked my hand a few times for wanting to have helper methods attached.  Go is nasty no matter what you do, so it didn't bother me much.